src/Entity/Geo/Canton.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace App\Entity\Geo;
+
+use Algolia\AlgoliaSearchBundle\Mapping\Annotation as Algolia;
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Entity
+ * @ORM\Table(name="geo_canton")
+ * @Algolia\Index(autoIndex=false)
+ */
+class Canton
+{
+    /**
+     * @var string
+     *
+     * @ORM\Id
+     * @ORM\Column
+     */
+    public $id;
+
+    /**
+     * @var string
+     *
+     * @ORM\Column
+     */
+    public $name;
+
+    /**
+     * @var Department
+     *
+     * @ORM\ManyToOne(targetEntity="App\Entity\Geo\Department")
+     * @ORM\JoinColumn(nullable=false)
+     */
+    public $department;
+
+    public function __construct(string $id, string $name, Department $department)
```

Comment:
we don't pass the id, we keep Doctrine manage it. (same for other constructs)

Reply:
https://github.com/EnMarche/en-marche.fr/pull/4584#discussion_r447715094